qiskit_nature/second_q/hamiltonians/heisenberg_model.py

```diff
@@ -23,21 +23,43 @@
 
 
 class HeisenbergModel(LatticeModel):
-    """The Heisenberg model."""
+    r"""The Heisenberg model.
+
+    This class implements the following Hamiltonian:
+
+    .. math::
+        H = - \vec{J} \sum_{\langle i, j \rangle} \vec{\sigma}_{i} \otimes \vec{\sigma}_{j}
+        - \vec{h} \sum_{i} \vec{\sigma}_{i}
+
+    where :math:`i,j` refer to lattice nodes. The :math:`\sum_{\langle i, j \rangle}` is performed
+    over adjacent lattice nodes. This model assumes spin-:math:`\frac{1}{2}` particles. Thus,
+    :math:`\vec{\sigma}_{i} = (X_i, Y_i, Z_i)` is a vector containing the Pauli matrices.
+    :math:`\vec{J}` is the coupling constant and :math:`\vec{h}` is the external magnetic field,
+    both with dimensions of energy.
+
+    This model is instantiated using a
+    :class:`~qiskit_nature.second_q.hamiltonians.lattices.Lattice`. For example, using a
+    :class:`~qiskit_nature.second_q.hamiltonians.lattices.LineLattice`:
+
+    .. code-block:: python
+
+        line_lattice = LineLattice(num_nodes=10, boundary_condition=BoundaryCondition.OPEN)
+        heisenberg_model = HeisenbergModel(line_lattice, (1.0, 1.0, 1.0), (0.0, 0.0, 1.0))
```

Comment:
Could you add a second example below this one, which shows how one can actually choose the model parameters to reproduce the `IsingModel` as a special case of this one? That example is contained in the release note which added this operator, here: https://github.com/Qiskit/qiskit-nature/blob/488e43e0754374823e9c98a16736bbf3a4583653/releasenotes/notes/added-heisenberg-model-218b7ca8386e208a.yaml#L20-L26